### PR TITLE
Listen to repair app event

### PR DESCRIPTION
**Pull request by VicDeo**
Implements #8

### CODE REVIEW

File: lib/Listener.php [hunk 1]
```diff
@@ -33,4 +33,8 @@ public function __construct(MarketService $marketService) {
 	public function upgradeAppStoreApp($app){
 		$this->marketService->updateApp($app);
 	}
+
+	public function repairAppStoreApp($app){
+		$this->marketService->installApp($app);
```
Comment (PVince81) [inline lib/Listener.php]: wow, that was easy 😄

**Comment by PVince81** [comment]
Requires https://github.com/owncloud/core/pull/27711 to be able to work

**Comment by PVince81** [comment]
Code looks good, will test a bit.

**Comment by PVince81** [comment]
Code looks good, tested together with https://github.com/owncloud/core/pull/27711.

Let's wait for that one to be merged and rerun the tests.

**Comment by PVince81** [comment]
#### Marketplace unavailable / offline
- [ ] TEST: upgrade with no changes to do: skipped, success
- [ ] TEST: upgrade with outdated app: abort
- [ ] TEST: upgrade with missing app code: abort
- [ ] TEST: admin manually updates local app code: success
- [ ] TEST: admin disables broken app: success

#### Marketplace available but app does not exist there
- [ ] TEST: upgrade with no changes to do: skipped, success
- [ ] TEST: upgrade with outdated app: abort
- [ ] TEST: upgrade with missing app code: abort
- [ ] TEST: admin manually updates local app code: success
- [ ] TEST: admin disables broken app: success

#### Marketplace available, app exists but no new version
- [ ] TEST: upgrade with no changes to do: skipped, success
- [ ] TEST: upgrade with outdated app: abort
- [ ] TEST: upgrade with missing app code: app is downloaded from market, success
- [ ] TEST: admin manually updates local app code: success
- [ ] TEST: admin disables broken app: success

#### Marketplace available, app exists and new version available
- [ ] TEST: upgrade with no changes to do: skipped, success
- [ ] TEST: upgrade with outdated app: app updated, success
- [ ] TEST: upgrade with missing app code: app downloaded from market, success
- [ ] TEST: upgrade still compatible app, and version is available: app updated, success
- [ ] TEST: admin manually updates local app code: success
- [ ] TEST: admin disables broken app: success

File: lib/MarketService.php [hunk 2]
```diff
@@ -85,6 +86,8 @@ public function installApp($appId) {
 			$package = $this->downloadPackage($appId);
 			$this->installPackage($package);
 			//Market is down or misfunctional
+		} catch (ClientException $e){
+			throw new AppManagerException('No marketplace connection');
```
Comment (butonic) [inline lib/MarketService.php]: can we not swallow the underlying exception here and in the other places:
```
throw new AppManagerException('No marketplace connection', 0, $e);
```

It is interesting to know if it was a Client or ServerException ... as well as the message they (shoud) contain

Comment (PVince81) [inline lib/MarketService.php]: Adjusted. I also adjusted the other PR to output it with the error message.

**Comment by PVince81** [comment]
rebased

**Comment by VicDeo** [comment]
@PVince81 I can't merge it here without approval

**Pull request merged**